scripts/tests/test_validate_reference_fixtures.py

```
@@ -0,0 +1,154 @@
+        manifests = [copy.deepcopy(self.manifest)]
+        dirty_revision = "a" * 40 + "-dirty"
+        raw["implementation"]["revision"] = dirty_revision
+        manifests[0]["implementation"]["source_revision"] = dirty_revision
+
+        with (
+            mock.patch.object(
+                validator,
+                "parse_key_value_record",
+                return_value={"implementation.revision": dirty_revision},
+            ),
+            self.assertRaisesRegex(ValueError, "clean Git commit"),
```

Ruff (0.15.21)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)